pygfx/utils/show.py

```diff
@@ -34,33 +44,47 @@ def show(object: WorldObject, up=None):
 
         background = Background(None, BackgroundMaterial((0, 1, 0, 1), (0, 1, 1, 1)))
         scene.add(background)
+        scene.add(AmbientLight(0.2))
 
-    camera = PerspectiveCamera(70, 16 / 9)
-    look_at = camera.show_object(object)
-    scene.add(camera)
+    if not camera:
+        camera = PerspectiveCamera(70, 16 / 9)
+        camera.add(DirectionalLight(0.8))
+        scene.add(camera)
 
-    # Add lights if there are none
-    light_count = 0
+    if canvas is None:
+        from wgpu.gui.auto import WgpuCanvas, run
 
-    def check_light(ob):
-        nonlocal light_count
-        if isinstance(ob, Light):
-            light_count += 1
+        canvas = WgpuCanvas()
+        event_loop = run
+    elif event_loop is None:
+        # TODO: find the matching run function
```

Comment:
This should do the trick:
```py
sys.modules[renderer.target.__module__].run
```

Reply:
Do you think this would mean we can drop the `event_loop` argument (since it comes with the canvas)? If not I propose a different name for `event_loop`, because it sounds like an object, which may be confusing.

Reply:
If we can find the `run` function for a given canvas then that would make it easier indeed. The question is if we want to allow people to experiment with their own event loop for a canvas, or if this is out of scope for now.

If it is out of scope, we can drop `event_loop` for now.

Reply:
Agree with dropping `event_loop` for now